XCGLogger/Library/XCGLogger/XCGLogger.swift

```diff
@@ -203,8 +206,9 @@ public class XCGNSLogDestination: XCGBaseLogDestination {
         else {
             adjustedText = text
         }
-
-        NSLog("%@", adjustedText)
+        dispatch_async(queue) {
+            NSLog("%@", adjustedText)
```

Comment:
Actually most logic could be run in the BG. The only logic that needs to remain on the main thread is the detection of the thread of a log event. At least for me it does not really matter on what thread the logging happened (since I could log that manually in the log string) but that is a design decision :)